### PR TITLE
Opacity refactor

### PR DESCRIPTION
**Pull request by Rohesie**
## About The Pull Request

* Moves all `opacity` var manipulation to a proc which sends a signal.
* light_blocker element for movable opaque atoms made, which tracks its movement and updates the affected turfs for proper lighting updates.
* `has_opaque_atom` boolean replaced by the `opacity_sources` lazylist to keep track of the sources, and a `directional_opacity` which serves a similar function but also allows for future expansion with on-border opaque objects (not yet implemented).
* Some opacity-related sight procs optimized as a result of this.
* Some variables moved to the object's definition.
* A define or two added into the mix for clarity.
* Some code cleaning, like turning booleans into their defines.
* One file renamed for clarity.

## Changelog
:cl:
balance: Mechs no longer block sight. It's a non-trivial cost for the lighting system with little to no gain.
/:cl:

### CODE REVIEW

File: code/modules/lighting/lighting_atom.dm [hunk 1]
```diff
@@ -41,25 +41,40 @@
...
 		return
 
-	if (new_opacity == TRUE)
-		T.has_opaque_atom = TRUE
-		T.reconsider_lights()
+	var/turf/turf_loc = loc
+
+	if(opacity)
+		AddElement(/datum/element/light_blocking)
+		turf_loc.add_opacity_source(src)
```
Comment (MrPerson) [inline code/modules/lighting/lighting_atom.dm:67]: I feel like this should be handled by the light_blocking element itself. Same for calling remove_opacity_source() upon removal of the element, obviously.

Comment (Rohesie) [reply code/modules/lighting/lighting_atom.dm]: Sure

**Comment by nicbn** [comment]
Could Mecha block sight but not light instead?

**Comment by Rohesie** [comment]
> 
> 
> Could Mecha block sight but not light instead?

Both use the same system currently, so no unless there's a big refactor and that'd be a bit out of scope.

**Pull request merged**